standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/DatabaseProduct.java

```diff
@@ -93,10 +100,6 @@ public static DatabaseProduct determineDatabaseProduct(String productName,
 
       // Check for null again in case of race condition
       if (theDatabaseProduct == null) {
-        Preconditions.checkNotNull(conf, "Configuration is null");
```

Comment:
Shouldn't we check this for race conditions/concurrent connections?

Reply:
I omitted the null configuration check in my original commit.  I have added it back but it was moved to line 80 right before it is needed before using the configuration to determine whether we have ConfVars.USE_CUSTOM_RDBMS set.